api/evaluation/evaluation.proto

```diff
@@ -0,0 +1,165 @@
+  optional string filtered_cloud_service_id = 1
+    [ (validate.rules).string.uuid = true ];
+
+  // Optional. Lists only evaluation results for a specific control id.
+  optional string filtered_control_id = 2
+    [(validate.rules).string.min_len = 1];
+
+  // Optional. Lists all evaluation results for the given initial control id substring, e.g., if the substring 'CMK-01.' is given it returns the controls CMK-01.1B, CMK-01.1S, CMK-01.1H.
+  optional string filtered_sub_controls = 3
+    [(validate.rules).string.min_len = 1];
+
+  // Optional. Latest results grouped by resource_id and metric_id.
+  optional bool latest_by_resource_id = 4;
+
+  int32 page_size = 10;
+  string page_token = 11;
+  string order_by = 12;
+  bool asc = 13;
+}
+
+message ListEvaluationResultsResponse {
+  repeated EvaluationResult results = 1;
+  string next_page_token = 2;
+}
+
+message StartEvaluationRequest {
+  string cloud_service_id = 1
+    [ (validate.rules).string.uuid = true ];
+  string catalog_id = 2
+    [ (validate.rules).string.min_len = 1 ]; 
+
+  // The interval time in minutes the evaluation executes periodically. The default interval is set to 5 minutes.
+  optional int32 interval = 3
+    [ (validate.rules).int32.gt = 0 ];
+}
+
+message StartEvaluationResponse {
+  bool status = 1; 
+  string status_message = 2; 
+}
+
+message StopEvaluationRequest {
+  string cloud_service_id = 1
+    [ (validate.rules).string.uuid = true ];
+
+  string catalog_id = 2
+    [ (validate.rules).string.min_len = 1 ]; 
+}
+
+message StopEvaluationResponse {}
+ 
+// A evaluation result resource, representing the result after evaluating the cloud service with a specific control
+// cloud_service_id, category_name and catalog_id are necessary to get the corresponding TargetOfEvaluation
+message EvaluationResult {
+  // Evaluation result id
+  string id = 1
+    [(validate.rules).string = {ignore_empty : true, uuid : true}]; 
+
+  // The Cloud Service ID the evaluation belongs to
+  string cloud_service_id = 2
+  [(validate.rules).string = {ignore_empty : true, uuid : true}]; 
+
+  // The control id the evaluation was based on
+  string control_id = 3
+    [(validate.rules).string.min_len = 1];
+  
+  // The category the evaluated control belongs to
+  string control_category_name = 4
+    [(validate.rules).string.min_len = 1]; 
+
+  // The catalog the evaluated control belongs to
+  string control_catalog_id = 5
+    [(validate.rules).string.min_len = 1]; 
+
+  // The evaluated resource 
+  string resource_id = 6
+    [(validate.rules).string.min_len = 1]; 
+  
+  // Evaluation status
+  EvaluationStatus status = 7 
+    [(validate.rules).enum.defined_only = true];
+  
+  
+  // Time of evaluation
+  google.protobuf.Timestamp timestamp = 8
+    [(tagger.tags) = "gorm:\"serializer:timestamppb;type:time\"" ];
+
+  // List of assessment results because of which the evaluation status is not 'compliant'
+  repeated string failing_assessment_result_ids = 9 [
```

Comment:
That would probably be too much. I would also propose just listing the failing ones here.

Reply:
It shouldn't be so much, because we are only using the last assessment results for the evaluation. I think it is a good idea to add the not failing assessment results as well. I could imagine that an auditor would want to see this information.

Reply:
For now, we do not add the not failling assessment results. 

Check the list creation.